lms/djangoapps/commerce/api/v1/serializers.py

```diff
@@ -17,6 +18,14 @@ def get_identity(self, data):
         except AttributeError:
             return None
 
+    def validate(self, data):
+        data = super(CourseModeSerializer, self).validate(data)
+        if 'professional' in data['mode_slug'] and data['expiration_datetime'] is not None:
```

Comment:
@wedaly good idea, will do that.  I had already started to move validation into the model code so that will dovetail nicely.